addons/addon-base-rest-api/packages/services/lib/authentication-providers/built-in-providers/cogito-user-pool/provider-service.js

```diff
@@ -168,18 +167,33 @@ class ProviderService extends Service {
           missingAttribs[key] = value;
         }
       });
+
+      // When IdP users are created via SWB UI, by default we use the username part of provided email address as first and last names
+      // To update these default names, we extract the mapped attribute values coming from the Cognito token
+      // Some IdPs send email value in their firstName and lastName attribute fields, so splitting string to ignore domain
+      const updateIfDifferent = attribName => {
+        if (
+          !_.isUndefined(userAttributes[attribName]) &&
+          existingUser[attribName] !== userAttributes[attribName].split('@')[0]
+        ) {
+          updatedAttribs[attribName] = userAttributes[attribName].split('@')[0];
```

Comment:
Why do we need the split on @? Do we expect Cognito or IdP values to have saved the names as emails? I did not observe that.

Reply:
There are external IdPs (like Auth0 dev) that attaches email in the firstname value.